oat-sa/extension-tao-mediamanager
# Feature/AUT-30/style editor for passage

Related to: https://oat-sa.atlassian.net/browse/AUT-30

**FE part:** 

- was added Style Editor. Reused code from taoQtiItem. 
- for supporting styles `attributes.class` is used `.tao-{hash}`

**BE part.**
- added endpoints for saving/loading CSS for passages
- fixed some storage related taoRevision compatibility issues
- store passages in directory when create/edit a passage

**What was not done**
- _Unit test will be added in scope of next ticket https://oat-sa.atlassian.net/browse/AUT-18_
- _Migration for existing passages will be done in separate PR_
- _support passage with styles is separate task https://oat-sa.atlassian.net/browse/AUT-35_
- _[BE] unit tests for `oat\taoMediaManager\model\sharedStimulus\css\service`_

**How to test:**
- go to Assets
- create new Passage (don't use existing they are not supported for now)
- go to Authoring and use style editor
- save
- open Authoring again
- check saved styles is applied
